AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSAnalyticsTransmissionTarget.m

```diff
@@ -37,10 +56,49 @@ - (MSAnalyticsTransmissionTarget *)transmissionTargetForToken:(NSString *)token
   // Look up for the token in the dictionary, create a new transmission target if doesn't exist.
   MSAnalyticsTransmissionTarget *target = self.childTransmissionTargets[token];
   if (!target) {
-    target = [[MSAnalyticsTransmissionTarget alloc] initWithTransmissionTargetToken:token];
+    target = [[MSAnalyticsTransmissionTarget alloc] initWithTransmissionTargetToken:token parentTarget:self];
     self.childTransmissionTargets[token] = target;
   }
   return target;
 }
 
+- (BOOL)isEnabled {
+
+  // Get isEnabled value from persistence.
+  // No need to cache the value in a property, user settings already have their cache mechanism.
+  NSNumber *isEnabledNumber = [self.storage objectForKey:self.isEnabledKey];
+
+  // Return the persisted value otherwise it's enabled by default.
+  return (isEnabledNumber) ? [isEnabledNumber boolValue] : YES;
+}
+
+- (void)setEnabled:(BOOL)isEnabled {
+  if (self.isEnabled != isEnabled) {
+
+    // Don't enable if the immediate parent is disabled.
+    if (isEnabled && ![self isAncestorEnabled]) {
+      MSLogWarning([MSAnalytics logTag],
+                   @"Can't enable; parent transmission target and/or Analytics service is disabled.");
+      return;
+    }
+
+    // Persist the enabled status.
+    [self.storage setObject:@(isEnabled) forKey:self.isEnabledKey];
+  }
+
+  // Propagate to nested transmission targets. TODO Find a more effective approach.
```

Comment:
Can probably found a better algorithm to travers the child tree but it will do the trick for now.

Reply:
You have do do a single operation for each child so it will still take O(N = members of child tree), I think.  We might be able to reduce recursion but this is probably fine.